_posts/2016-01-29-CocoaPods-1.0.markdown

```diff
@@ -0,0 +1,140 @@
+
+``` ruby
+source 'https://github.com/CocoaPods/Specs.git'
+use_frameworks!
+
+target 'ORStackView_Example' do
+  pod "ORStackView", :path => "../"
+
+  target 'ORStackView_Tests' do
+    inherit! :search_paths
+
+    pod 'Quick'
+  end
+end
+```
+
+It has an obvious hierarchy, so you can see that Pods included in `ORStackView_Example` will be included in the `ORStackView_Tests` target, the only new thing is `inherit! :search_paths` which means "don't link Pods into here, but let this target know they exist." This `Podfile` is easier to read, simpler and has less redundant information.
+
+### Off with `pod`s `:head`
+
+Adding `:head` to a `pod` nearly always worked, except when it didn't. It is documented as this:
+
+> Finally, instead of a version, you can specify the :head flag. This will use the pod’s latest version spec version, but force the download of the ‘bleeding edge’ version. Use this with caution, as the spec might not be compatible with the source material anymore.
+
+It was a tool that was really useful back when CocoaPods was small and it was hard to persuade people it was worth support any dependency manager for their projects. That time has long passed and now the flexibility that this attribute originally offered has fallen onto the side of "not worth keeping around."It would introduce extremely hard to debug issues for developers expecting it to be "use the latest version of their code repo." You can either use the `:podspec` option or point directly to the upstream repo ( via `git`, `:svn`, etc. )
+
+
+### Options to `install!`
+
+In the past when we have had to support configuration options for CocoaPods we have used environment variables. These act like NSUserDefaults in that someone who knows the key can make a change, but otherwise they remain [quite hidden](https://github.com/ChatSecure/ChatSecure-iOS/commit/d766400c2d8e935ab49daf5ef5f9bf28c760e484). Though, sometimes they could be command line APIs too like `--no-integrate`. These were a conscious choices that eventually added up to being inconsistent over time, and we've opted to move away from that to centralizing the information in the Podfile.
+
+This means that every developer on your team would have the same output on `pod install` regardless of their environment. Whether in [CocoaPods.app](https://cocoapods.org/app) or in the command line. We've left the DSL attribute `install!` quite open so that [future features](https://github.com/CocoaPods/CocoaPods/issues/2729) can be built with this attribute.
+
+### Building for stability
+
+A lot of the work for 1.0 has been around trying to simplify CocoaPods in order to provide long-lasting infrastructure. This is especially true given that the foundations that CocoaPods builds upon often shifts [every June](https://en.wikipedia.org/wiki/Apple_Worldwide_Developers_Conference) and the project is ran by a handful of people in their spare time.
```

Comment:
![dave-gibson-and-charlie-talbot-battle-of-britain-north-vs-south_22140](https://cloud.githubusercontent.com/assets/49038/12716328/7ad9e034-c8d7-11e5-819f-f6aa8000f42c.jpg)

but yeah, my N.Irish girlfriend would tell me to change that, so I should do it :dancers: